SConstruct

```diff
@@ -360,9 +360,15 @@ config_options = [
         """Enable collection of code coverage information with gcov.
            Available only when compiling with gcc.""",
         False),
+    BoolOption(
+        "doxygen_xml",
+        "Build XML tree for the C++ interface using Doxygen.",
+        False),
```

Comment:
I'm mildly concerned about the additional complexity having an additional option here introduces. Is there a problem with generating HTML and XML simultaneously? Is it substantially slower or anything?

Reply:
The HTML step is somewhat slow, but not excessively so. The CI uses the XML as artifacts in later steps (e.g., the `dotnet` runner but I anticipate the same for future MATLAB runners). If it helps, I can switch the defaults so both XML and HTML are built by default, but HTML output can be disabled.

Fwiw, some of this is a test for future modifications of the Doxygen output to document CLib itself (i.e., run `scons doxygen` for XML, generate CLib, and then run `scons doxygen` again to get the full HTML for both C++ API and CLib API).